api/download.py

```diff
@@ -84,12 +86,12 @@ def _bulk_preflight_archivestream(self, file_refs):
         targets = []
 
         for fref in file_refs:
-            cont_name   = fref.get('container_name','')+'s'
+            cont_name   = pluralize(fref.get('container_name',''))
```

Comment:
This would be the try/catch location (or modify the if statement below to be singular and check first)